substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1880,6 +1881,28 @@ impl<T: Config> StakingInterface for Pallet<T> {
 		}
 	}
 
+	fn force_withdraw(stash: &Self::AccountId, amount: Self::Balance) -> DispatchResult {
+		let mut ledger = Self::ledger(Stash(stash.clone()))?;
+		ledger.active = ledger.active.saturating_sub(amount);
+		// FIXME eagr should we check if active >= ED before updating?
+		ledger.total = ledger.total.saturating_sub(amount);
+		ledger.update()?;
+
+		if T::VoterList::contains(stash) {
+			let _ = T::VoterList::on_update(stash, Self::weight_of(stash)).defensive();
+		}
+
+		// HACK Delegation relies on the `on_withdraw` event to increase agent's unclaimed
+		// withdrawals. We do this to maintain the ledger integrity.
+		T::EventListeners::on_withdraw(stash, amount);
+		Self::deposit_event(Event::<T>::Withdrawn { stash: stash.clone(), amount });
+
+		T::Currency::burn(amount);
```

Comment:
seems weird to withdraw as well as burn? Shouldn't it be nomination pool to decide what to do with withdrawn funds?

Reply:
> Shouldn't it be nomination pool to decide what to do with withdrawn funds?

Yeah, that makes more sense :))